lib/server/services/qualityControlFlag/QcFlagService.js

```diff
@@ -344,32 +359,44 @@ class QcFlagService {
 
                 const createdFlag = await QcFlagRepository.findOne(this.prepareQueryBuilder().where('id').is(newInstance.id));
 
-                const flagInclude = [];
+                const flagIncludes = [];
                 let synchronousQcWhereClause = null;
                 if (dataPass) {
                     await createdFlag.addDataPass(dataPass);
-                    flagInclude.push({ association: 'dataPasses', where: { id: dataPass.id }, required: true });
+                    flagIncludes.push({ association: 'dataPasses', where: { id: dataPass.id }, required: true });
                 } else if (simulationPass) {
                     await createdFlag.addSimulationPass(simulationPass);
-                    flagInclude.push({ association: 'simulationPasses', where: { id: simulationPass.id }, required: true });
+                    flagIncludes.push({ association: 'simulationPasses', where: { id: simulationPass.id }, required: true });
                 } else {
-                    flagInclude.push({ association: 'dataPasses', required: false });
-                    flagInclude.push({ association: 'simulationPasses', required: false });
+                    flagIncludes.push({ association: 'dataPasses', required: false });
+                    flagIncludes.push({ association: 'simulationPasses', required: false });
                     synchronousQcWhereClause = { '$flag->dataPasses.id$': null, '$flag->simulationPasses.id$': null };
                 }
                 // Else synchronous QC flag is created
 
                 const effectivePeriodsToBeUpdated = await QcFlagEffectivePeriodRepository.findAll({
                     subQuery: false,
                     where: {
-                        from: { [Op.lt]: createdFlag.to },
-                        to: { [Op.gt]: createdFlag.from },
+                        [Op.and]: [
+                            {
+                                [Op.or]: [
+                                    { from: null },
+                                    to ? { from: { [Op.lt]: to } } : { from: { [Op.not]: null } },
+                                ],
```

Comment:
I might be wrong, but this looks like if `to` is falsy, you have 
```js
[Op.or]: [
    { from: null },
    { from: { [Op.not]: null } },
]
```
Which is aways true, so maybe you should build your where criterias and having something like
```js
whereCriteria = [];
if (to) {
    whereCriteria.push({
        [Op.or]: [
            { from: null },
            { from: { [Op.lt]: to } }
        ]
    })
}
// same for from
// and then you have
{
    where: { [Op.and]: whereCriteria },
}
```

Reply:
applied